File: app/src/main/java/com/whatsapp/stickerimporter/StickerPackAdapter.java
package com.whatsapp.stickerimporter;

import android.content.Context;
import android.graphics.BitmapFactory;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

/**
 * Adapter for displaying sticker packs in RecyclerView
 */
public class StickerPackAdapter extends RecyclerView.Adapter<StickerPackAdapter.ViewHolder> {

    private final Context context;
    private List<StickerPack> stickerPacks;
    private final OnStickerPackClickListener listener;

    public interface OnStickerPackClickListener {
        void onAddToWhatsApp(StickerPack stickerPack);
        void onDelete(StickerPack stickerPack);
        void onViewDetails(StickerPack stickerPack);
    }

    public StickerPackAdapter(Context context, List<StickerPack> stickerPacks, OnStickerPackClickListener listener) {
        this.context = context;
        this.stickerPacks = new ArrayList<>(stickerPacks);
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_sticker_pack, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        StickerPack stickerPack = stickerPacks.get(position);

        holder.packName.setText(stickerPack.name);
        holder.packInfo.setText(stickerPack.stickers.size() + " stickers • " +
            formatFileSize(stickerPack.totalSize));

        // Load tray icon
        File packDir = new File(context.getFilesDir(), "sticker_packs/" + stickerPack.identifier);
        File trayFile = new File(packDir, stickerPack.trayImageFile);
        if (trayFile.exists()) {
            holder.trayIcon.setImageBitmap(BitmapFactory.decodeFile(trayFile.getAbsolutePath()));
        } else {
            holder.trayIcon.setImageResource(R.drawable.ic_launcher);
        }

        holder.addButton.setOnClickListener(v -> {
            if (listener != null) {
                listener.onAddToWhatsApp(stickerPack);
            }
        });

        holder.deleteButton.setOnClickListener(v -> {
            if (listener != null) {
                listener.onDelete(stickerPack);
            }
        });

        holder.itemView.setOnClickListener(v -> {
            if (listener != null) {
                listener.onViewDetails(stickerPack);
            }
        });
    }

    @Override
    public int getItemCount() {
        return stickerPacks.size();
    }

    public void updateStickerPacks(List<StickerPack> newPacks) {
        this.stickerPacks = new ArrayList<>(newPacks);
        notifyDataSetChanged();
    }

    private String formatFileSize(long bytes) {
        if (bytes < 1024) {
            return bytes + " B";
        } else if (bytes < 1024 * 1024) {
            return String.format("%.1f KB", bytes / 1024.0);
        } else {
            return String.format("%.1f MB", bytes / (1024.0 * 1024.0));
        }
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView trayIcon;
        TextView packName;
        TextView packInfo;
        Button addButton;
        Button deleteButton;

        ViewHolder(View itemView) {
            super(itemView);
            trayIcon = itemView.findViewById(R.id.trayIcon);
            packName = itemView.findViewById(R.id.packName);
            packInfo = itemView.findViewById(R.id.packInfo);
            addButton = itemView.findViewById(R.id.addButton);
            deleteButton = itemView.findViewById(R.id.deleteButton);
        }
    }
}

File: app/src/main/java/com/mystickers/whatsapp/StickerPackAdapter.kt
package com.mystickers.whatsapp

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import android.graphics.BitmapFactory

/**
 * RecyclerView adapter for displaying sticker packs.
 */
class StickerPackAdapter(
    private val packs: List<StickerPack>,
    private val onAddToWhatsApp: (StickerPack) -> Unit
) : RecyclerView.Adapter<StickerPackAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val trayIcon: ImageView = view.findViewById(R.id.tray_icon)
        val packName: TextView = view.findViewById(R.id.pack_name)
        val packInfo: TextView = view.findViewById(R.id.pack_info)
        val addButton: Button = view.findViewById(R.id.btn_add_to_whatsapp)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_sticker_pack, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val pack = packs[position]
        holder.packName.text = pack.name
        holder.packInfo.text = "${pack.stickers.size} Sticker · ${pack.publisher}"

        // Load tray icon
        val context = holder.itemView.context
        val manager = StickerPackManager.getInstance(context)
        val trayFile = manager.getTrayIconFile(pack.identifier)
        if (trayFile != null && trayFile.exists()) {
            val bitmap = BitmapFactory.decodeFile(trayFile.absolutePath)
            holder.trayIcon.setImageBitmap(bitmap)
        }

        holder.addButton.setOnClickListener {
            onAddToWhatsApp(pack)
        }
    }

    override fun getItemCount(): Int = packs.size
}
